File: ArraysAndHashing/FirstMissingPositive.py
class Solution(object):
    def firstMissingPositive(self, nums):
        s = set()
        n = len(nums)
        for i in range(n):
            if nums[i] > 0 and nums[i] <= n:
                s.add(nums[i])
        for i in range(1,n+2):
            if i not in s:
                return i
        

File: ArraysAndHashing/FirstMissingPositive.cpp
#include <vector>
#include <set>

using namespace std;

class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        set<double> numsSet(nums.begin(),nums.end());
        for(int i = 1; i < nums.size()+1 ; i++)
        {
            if(numsSet.find(i) == numsSet.end())
                return i;
        }
        return nums.size()+1;
    }
};